src/ovp_pipeline/truth_api.py

```
@@ -1262,51 +1268,110 @@ def list_objects(
+            " o.canonical_path AS canonical_path, o.source_slug AS source_slug,"
+            " COALESCE(r.backlinks, 0) AS backlink_count,"
+            " ROW_NUMBER() OVER ("
+            "PARTITION BY o.object_id"
+            f" ORDER BY CASE o.pack {pack_order} ELSE {fallback_order} END"
+            ") AS rn"
+            " FROM objects o"
+            " LEFT JOIN ("
+            "SELECT pack, target_object_id, COUNT(*) AS backlinks"
+            " FROM relations GROUP BY pack, target_object_id"
+            ") r ON r.pack = o.pack AND r.target_object_id = o.object_id"
+            f" WHERE {where_clause}"
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)